packages/protocol/src/protocol/MessageChannel.ts

```diff
@@ -0,0 +1,161 @@
+import { Message, MessageCodec } from "#codec/index.js";
+import { Channel, Logger, MatterFlowError } from "#general";
+import { DEFAULT_EXPECTED_PROCESSING_TIME_MS, ExchangeLogContext } from "#protocol/MessageExchange.js";
+import { Session, SessionParameters } from "#session/index.js";
+
+const logger = new Logger("MessageChannel");
+
+/**
+ * The buffer time in milliseconds to add to the peer response time to also consider network delays and other factors.
+ * TODO: This is a pure guess and should be adjusted in the future.
+ */
+const PEER_RESPONSE_TIME_BUFFER_MS = 5_000;
+
+/**
+ * The maximum number of transmission attempts for a given reliable message. The sender MAY choose this value as it
+ * sees fit.
+ */
+export const MRP_MAX_TRANSMISSIONS = 5;
```

Comment:
This is NBD but if you created an "MRP" namespace you could do `MRP.MAX_TRANSMISSIONS`...  Would be same number of characters and create less export noise